hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NodeSchemaManager.java

```diff
@@ -55,6 +58,18 @@ public static NodeSchemaManager getInstance() {
     return instance;
   }
 
+  public static NodeSchemaManager getInitiatedInstance(String schemaFile) {
+    NodeSchemaManager cachedInstance = cache.get(schemaFile);
+    if (cachedInstance == null) {
+      instance = new NodeSchemaManager();
+      instance.init(schemaFile);
+      cache.put(schemaFile, instance);
+      return instance;
+    } else {
+      return cachedInstance;
+    }
```

Comment:
This is not thread-safe.  Use `cache.computeIfAbsent()` if `NodeSchemaManager.init()` has no side-effects.

Reply:
Thanks, will fix.